Assets/PatchKit Patcher/Scripts/UI/DownloadSpeed.cs

```diff
@@ -11,10 +13,35 @@ private void Start()
         {
             Patcher.Instance.UpdateAppStatusChanged += status =>
             {
-                Text.text = status.IsDownloading ? (status.DownloadBytesPerSecond / 1024.0).ToString("0.0 kB/sec.") : string.Empty;
+                string speed = (status.DownloadBytesPerSecond / 1024.0).ToString("0.0 kB/sec.");
+                string eta = (status.DownloadBytes > 0 && status.DownloadBytes < status.DownloadTotalBytes)
```

Comment:
Why not use `status.IsDownloading`?

Reply:
eta label was appearing sometimes shortly before and after actual download with wrong values ex. NaN